azure/ConsiderationsForServiceDesign.md

```diff
@@ -141,94 +142,196 @@ cannot collide with a resource path that contains user-specified resource ids.
-Here we illustrate the sequence of API calls involved in each of these patterns.
+1. The client sends the request to initiate the long-running operation.
+The initial request could be a PUT, POST, or DELETE method.
+PATCH must never be used for LROs -- it should be reserved for simple resource updates.
+If a long-running update is required it should be implemented with POST.
+The request may contain an `operation-id` header that the service uses as the ID of the status monitor created for the operation.
 
-### Resource-based long-running operations
+2. The service validates the request and initiates the operation processing.
+It sends a response to the client with a `202-Accepted` HTTP status code.
+The response body is the status monitor for the operation including the ID, either from the request header or generated by the service.
+When returning a status monitor whose status is not in a terminal state, the response must also include a `retry-after` header indicating the minimum number of seconds the client should wait 
+before polling (GETting) the status monitor URL again for an update.
+For backward compatibility, the response may also include an `Operation-Location` header containing the absolute URL
+of the status monitor resource (without an api-version query parameter).
 
-In the RELO pattern, the resource that is the target of the operation contains a `status` field
-that holds the status of an outstanding or last completed operation.
-This means that the client can use a standard "get" operation on the resource to determine the
-status of an operation it initiated. The flow looks like this:
+3. After waiting at least the amount of time specified by the previous response's `Retry-after` header,
+the client issues a GET request to the status monitor using the ID in the body of the initial response.
+The GET operation for the status monitor is documented in the REST API definition and the ID
+is the last URL path segment.
 
-<!-- markdownlint-disable MD033 -->
-<p align="center">
-  <img src="./relo.jpg" alt="The RELO flow"/>
-</p>
-<!-- markdownlint-enable MD033 -->
+4. The status monitor responds with information about the operation including its current status,
+which should be represented as one of a fixed set of string values in a field named `status`.
+If the operation is still being processed, the status field will contain a "non-terminal" value, like `NotStarted` or `Running`.
 
-1. The client sends the initial request to the resource to initiate the long-running operation.
-This initial request could be a PUT, PATCH, POST, or DELETE method.
+5. After the operation processing completes, a GET request to the status monitor returns the status monitor with a status field set to a terminal value -- `Succeeded`, `Failed`, or `Canceled` -- that indicates the result of the operation.
+If the status is `Failed` or `Canceled`, the status monitor resource contains an `error` field with a `code` and `message` that describes the failure.
```

Comment:
The LRO "failed" to complete due to cancellation. So, since the LRO is not successful, I think the error should indicate that it failed due to cancellation.

Reply:
I discussed with Jeffrey today and he is willing to reconsider, so we'll drop cancelled for now and see if we get feedback on this.